src/Pages/AuthPage/ProfileComponents/SignupBox/SignupBox.jsx

```diff
@@ -0,0 +1,191 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import { useAuthContext } from "../../../../Context/Auth.context";
+import { AuthContainer } from "../AuthContainer/AuthContainer";
+import { LoginBox } from "../LoginBox/LoginBox";
+
+export const SignupBox = () => {
+  const { signup } = useAuthContext();
+  const [userDetails, setUserDetails] = useState({
+    user: "",
+    email: "",
+    passwordOne: "",
+    passwordTwo: "",
+    terms: false,
+  });
+  const [error, setError] = useState({ state: false, text: "" });
+  const [type, setType] = useState(true);
+
+  const validation = /^(?=.*\d)(?=.*[a-z]).{5,10}$/;
+  const emailValidation =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
Its awesome that you used regex for email validation

Reply:
Thanks!